frontend/src/app/(pages)/profile/page.tsx

```diff
@@ -1,9 +1,69 @@
+"use client"
 import ProfileComponent from "@/components/profile/Profile"
+import { UserService } from "@/helpers/user/user_api_wrappers";
+import User from "@/types/user";
+import { Metadata } from "next"
+import { FormEvent, useEffect, useState } from "react";
+import { toast } from "react-toastify";
 
-export default function ProfilePage() {
+export const metadata: Metadata = {
+    title: "Profile",
+    description: "profile information",
+}
+
+export default async function ProfilePage() {
+
+    const [user, setUser] = useState<User | undefined>(undefined);
+
+    const email = typeof window !== "undefined" ? sessionStorage.getItem("email") || "" : "";
```

Comment:
Hmm, if I understand this line of code correctly, you are trying to check if the user is using web browser to access this page, and if this is true, you try getting the email from the web session. And will assign empty string if no email is retrievable from the session or the user is not using the web browser.
If my understanding is correct, you can simplify this to:
```
const email = sessionStorage.getItem("email") ?? "";
```
If the user is not using a browser, sessionStorage.getItem("email") returns null. Btw, if you don't like coleascing, feel free to replace it with ternary.

Reply:
i couldnt get the `auth.tsx` context to work here because i wasn't sure how to update the context after signing in and i wouldnt have used sessionStorage. do you know how to update the dummy auth context we have now from the login component? i can continue trying tonight

Reply:
Oh, I am also not familiar with React context... Ok then I am fine with you current implementation haha!

Reply:
I'm not familiar with context oso, but hope this helps...
```ts
// seems like you will need to export the user in the value field of your provider
<AuthContext.Provider value={{ isAuthenticated: !!user, user, login, loading, logout }}>
            {children}
</AuthContext.Provider>

// then the children can get value from the context.
const { user, loading } = useAuthContext();
```
see this https://www.mikealche.com/software-development/how-to-implement-authentication-in-next-js-without-third-party-libraries, they also use useSWR to wait for user data before fetching other infos.